dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java

```diff
@@ -414,5 +419,63 @@ String getURLQueryStringFromRequest(HttpServletRequest request)
 				: request.getQueryString();
 	}
 
+	/**
+	 * Verifies if a given user has READ permissions on a given permissionable object, if there is
+	 * no logged in user and the permissionable object requires authentication a 401 response error
+	 * is set in order to redirect the user to the login page and if an user is already logged in
+	 * but that logged in user does not have read permissions on the given permissionable object a
+	 * 403 response error is set.
+	 *
+	 * @param permissionable Permissionable object to validate
+	 * @param requestedURIForLogging The original requested URI, this URI is required as a parameter
+	 * just for logging purposes, is not used to calculate anything.
+	 * @param user Current user
+	 */
+	public Boolean isUnauthorizedAndHandleError(Permissionable permissionable,
+			String requestedURIForLogging, User user,
+			HttpServletRequest request, HttpServletResponse response)
+			throws IOException, DotDataException {
+
+		PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+		// Check if the page is visible by a CMS Anonymous role
+		if (!permissionAPI
+				.doesUserHavePermission(permissionable, PERMISSION_READ, user, true)) {
+
+			if (null == user) {//Not logged in user
+
+				Logger.debug(this.getClass(),
+						"CHECKING PERMISSION: Page doesn't have anonymous access ["
+								+ requestedURIForLogging + "]");
+				Logger.debug(this.getClass(), "401 URI = " + requestedURIForLogging);
+				Logger.debug(this.getClass(), "Unauthorized URI = " + requestedURIForLogging);
+
+				request.getSession().setAttribute(
+						com.dotmarketing.util.WebKeys.REDIRECT_AFTER_LOGIN, requestedURIForLogging);
+				response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
+						"The requested page/file is unauthorized");
+				return true;
+			} else if (!permissionAPI.getRolesWithPermission(permissionable, PERMISSION_READ)
+					.contains(APILocator.getRoleAPI().loadLoggedinSiteRole())) {
+				// User is logged in need to check user permissions
+				if (!permissionAPI
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)